core/src/components/list/list.ios.scss

```diff
@@ -19,12 +19,25 @@
   @include border-radius($list-inset-ios-border-radius);
 }
 
-.list-ios.list-inset ion-item:last-child {
+/**
+ * These selectors ensure the first item in the list
+ * has the correct border.
+ * We need to consider the following scenarios:
+ 1. The only item in a list.
+ 2. The last item in a list as long as it is not the only item.
+ 3. The item in the last item-sliding in a list.
+ * Note that we do not select ion-item:last-of-type
+ * because that will cause the borders to disappear on
+ * items in an item-sliding when the item is the last
+ * element in the item-sliding container.
+ */
+.list-ios.list-inset ion-item:only-child,
```

Comment:
Shouldn't this be `:only-of-type`? The other lines in this rule use `-type` selectors instead of `-child` ones, implying that we still want to remove the border even if there are other, non-item elements under the last item. Ditto for the other instances of this pattern.

Reply:
This rule is targeting when there is only 1 item inside of a list (Regardless of item-sliding usage). Doing `ion-item:only-of-type` causes lines to disappear when the item is using in `ion-item-sliding`.

Example: 
![image](https://github.com/ionic-team/ionic-framework/assets/2721089/65393afe-5fe5-4ff8-a2c3-0b4e30b7dc64)

Markup:

```html
<ion-list inset="true">
  <ion-item-sliding>
    <ion-item-options slot="start">
      <ion-item-option color="warning">
        <ion-icon slot="icon-only" name="pin"></ion-icon>
      </ion-item-option>
    </ion-item-options>
    <ion-item>
      <ion-label>Item 2</ion-label>
    </ion-item>
  </ion-item-sliding>
</ion-list>
```

Reply:
I created a Codepen of this if it helps you visualize it better: https://codepen.io/brandyscarney/pen/MWLewmr

Reply:
Ahh, makes sense 👍

Reply:
I added a comment in https://github.com/ionic-team/ionic-framework/pull/28439/commits/448b88c0a29cccb8989d2caea79a60522d0abdb4 to clarify